chain/cosmos/sidecar.go

```diff
@@ -0,0 +1,172 @@
+package cosmos
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	dockerclient "github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/strangelove-ventures/interchaintest/v7/ibc"
+	"github.com/strangelove-ventures/interchaintest/v7/internal/dockerutil"
+	"go.uber.org/zap"
+)
+
+type SidecarProcesses []*SidecarProcess
+
+// SidecarProcess represents a companion process that may be required on a per chain or per validator basis.
+type SidecarProcess struct {
+	log *zap.Logger
+
+	Index int
+	Chain ibc.Chain
+
+	// If true this process is scoped to a specific validator, otherwise it is scoped at the chain level.
+	validatorProcess bool
+
+	// If true this process should be started before the chain or validator, otherwise it should be explicitly started after.
+	preStart bool
+
+	// Set to true in StartContainer call so that we do not accidentally kick off the same process more than once.
+	started bool
```

Comment:
maybe instead of this flag, this could be a method implemented in the ContainerLifecycle that inspects docker for the running container ID?

Reply:
much better idea! just made changes to account for this